### DISCUSSION TITLE
error.tsx not catching error

### DISCUSSION THREAD

**Discussion by JacobWeisenburger** [vercel/next.js#51372] [Help]
### Summary

When I'm in dev mode, `next dev`, and my page throws an error, `error.tsx` seems to be working correctly.

![image](https://github.com/vercel/next.js/assets/31667350/7e331f9b-5b8d-4c1c-abf2-411c85995423)

However, when I try to do `next build` and then `next start`, `error.tsx` doesn't catch the error.

![image](https://github.com/vercel/next.js/assets/31667350/c8953a79-93f5-4941-8ac7-12b27996222b)

Is there something that I'm doing wrong?

`src\app\sandbox\[id]\page.tsx`
```tsx
const paths = [
    { id: '1' },
    { id: '2' },
    { id: '3' },
]
export function generateStaticParams () {
    return paths
}

export default function Page ( props: { params: { id: string } } ) {
    const path = paths.find( path => path.id == props.params.id )
    if ( !path ) throw new Error( `Not found: /sandbox/${ props.params.id }` )

    return <h1>{props.params.id}</h1>
}
```

`src\app\sandbox\[id]\error.tsx`
```tsx
'use client'

export default function Error ( props: { error: Error } ) {
    return <>
        <p>{props.error.message}</p>
    </>
}
```

**Comment by singleseeker** [comment]
Did you get the solution?
